tests/test_download.py

```diff
@@ -0,0 +1,113 @@
+from luminoso_api.v5_client import LuminosoClient
+from luminoso_api.v5_download import iterate_docs, download_docs
+
+import pytest
+import requests
+import tempfile
+import json
+
+BASE_URL = 'http://mock-api.localhost/api/v5/'
+RESPONSE = {
+    'result': [
+        {
+            'title': 'Document 1',
+            'text': 'hello',
+            'tokens': [{'term_id': 'hello|en'}],
+            'metadata': [],
+            'vector': 'AAAA',
+            'doc_id': 'uuid-0',
+        },
+        {
+            'title': 'Document 2',
+            'text': 'hello',
+            'tokens': [{'term_id': 'hello|en'}],
+            'metadata': [],
+            'vector': 'AAAA',
+            'doc_id': 'uuid-1',
+        },
+    ],
+    'total_count': 2,
+}
+
+EXPANDED_DOCS = [
+    {
+        'title': 'Document 1',
+        'text': 'hello',
+        'tokens': [{'term_id': 'hello|en'}],
+        'metadata': [],
+        'vector': 'AAAA',
+    },
+    {
+        'title': 'Document 2',
+        'text': 'hello',
+        'tokens': [{'term_id': 'hello|en'}],
+        'metadata': [],
+        'vector': 'AAAA',
+    },
+]
+
+CONCISE_DOCS = [
+    {'title': 'Document 1', 'text': 'hello', 'metadata': []},
+    {'title': 'Document 2', 'text': 'hello', 'metadata': []},
+]
+
+REPETITIVE_DOC = {'title': 'Yadda', 'text': 'yadda yadda', 'metadata': []}
+
+
+def test_iteration(requests_mock):
```

Comment:
I'm confused about where `requests_mock` is coming from in this test, and in `test_v5_client.py`.

Reply:
Yep. It comes from pytest; the name of the argument to your test function specifies a plugin that it needs, and passes in that plugin as the value. I could add a comment about it because pytest is new to us.